pkg/cloudprovider/aws/ami.go

```diff
@@ -83,14 +84,29 @@ func (p *AMIProvider) getAMIID(ctx context.Context, query string) (string, error
 	return ami, nil
 }
 
-func (p *AMIProvider) getSSMQuery(instanceType cloudprovider.InstanceType, version string) string {
+func (p *AMIProvider) getSSMQuery(ctx context.Context, constraints *v1alpha1.Constraints, instanceType cloudprovider.InstanceType, version string) string {
 	var amiSuffix string
+	var arch string
 	if !instanceType.NvidiaGPUs().IsZero() || !instanceType.AWSNeurons().IsZero() {
 		amiSuffix = "-gpu"
 	} else if instanceType.Architecture() == v1alpha5.ArchitectureArm64 {
 		amiSuffix = "-arm64"
+		arch = "arm64"
+	} else if instanceType.Architecture() == v1alpha5.ArchitectureAmd64 {
+		arch = "arm64"
```

Comment:
this should be `x86_64` (https://docs.aws.amazon.com/eks/latest/userguide/retrieve-ami-id-bottlerocket.html#:~:text=aws%2Dk8s%2D1.21/-,x86_64,-/latest/image_id%20%2D%2Dregion)

Reply:
This might be useful `v1alpha5.AWSToKubeArchitectures`

Reply:
^ `v1alpha1.AWSToKubeArchitectures`

Reply:
I don't totally understand this - isn't the default from SSM x86-64? What's the purpose of the new arch variable in the suggestion (vs the existing amiSuffix)?

Reply:
Fixed.